Source/Global/perform_env.cpp

```diff
@@ -262,6 +265,10 @@ HRESULT CALLBACK HC_PERFORM_ENV::HttpPerformAsyncShimProvider(XAsyncOp op, const
         RETURN_IF_FAILED(XTaskQueueGetPort(data->async->queue, XTaskQueuePort::Work, &workPort));
         RETURN_IF_FAILED(XTaskQueueCreateComposite(workPort, workPort, &performContext->internalAsyncBlock.queue));
 
+        std::unique_lock<std::mutex> lock{ env->m_mutex };
```

Comment:
why not use a scope to unlock?

Reply:
I can change.  Is that better somehow or just style preference?

Reply:
it's more obvious what is protected by the lock and what isn't

on the other hand I see a lot of code here using `.unlock()` so it may not be worthwhile to change

Reply:
keeping as is for consistency